instrumentation/micrometer/micrometer-1.5/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/micrometer/v1_5/LongTaskTimerTest.java

```diff
@@ -0,0 +1,217 @@
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.micrometer.v1_5;
+
+import static io.opentelemetry.sdk.testing.assertj.MetricAssertions.assertThat;
+import static io.opentelemetry.sdk.testing.assertj.OpenTelemetryAssertions.attributeEntry;
+
+import io.micrometer.core.instrument.LongTaskTimer;
+import io.micrometer.core.instrument.Metrics;
+import io.opentelemetry.instrumentation.testing.junit.AgentInstrumentationExtension;
+import io.opentelemetry.instrumentation.testing.junit.InstrumentationExtension;
+import java.time.Duration;
+import java.util.concurrent.TimeUnit;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+class LongTaskTimerTest {
+
+  static final String INSTRUMENTATION_NAME = "io.opentelemetry.micrometer-1.5";
+
+  @RegisterExtension
+  static final InstrumentationExtension testing = AgentInstrumentationExtension.create();
+
+  @BeforeEach
+  void cleanupMeters() {
+    Metrics.globalRegistry.forEachMeter(Metrics.globalRegistry::remove);
+  }
+
+  @Test
+  void testLongTaskTimer() throws Exception {
+    // given
+    LongTaskTimer timer =
+        LongTaskTimer.builder("testLongTaskTimer")
+            .description("This is a test long task timer")
+            .tags("tag", "value")
+            .register(Metrics.globalRegistry);
+
+    // when
+    LongTaskTimer.Sample sample = timer.start();
+
+    // then
+    testing.waitAndAssertMetrics(
+        INSTRUMENTATION_NAME,
+        "testLongTaskTimer.active",
+        metrics ->
+            metrics.anySatisfy(
+                metric ->
+                    assertThat(metric)
+                        .hasDescription("This is a test long task timer")
+                        .hasUnit("tasks")
+                        .hasLongSum()
+                        .isNotMonotonic()
+                        .points()
+                        .satisfiesExactly(
+                            point ->
+                                assertThat(point)
+                                    .hasValue(1)
+                                    .attributes()
+                                    .containsOnly(attributeEntry("tag", "value")))));
+
+    // when
+    TimeUnit.MILLISECONDS.sleep(100);
+    sample.stop();
+
+    // then
+    testing.waitAndAssertMetrics(
+        INSTRUMENTATION_NAME,
+        "testLongTaskTimer",
+        metrics ->
+            metrics.anySatisfy(
+                metric ->
+                    assertThat(metric)
+                        .hasDescription("This is a test long task timer")
+                        .hasUnit("ms")
+                        .hasDoubleHistogram()
+                        .points()
+                        .satisfiesExactly(
+                            point ->
+                                assertThat(point)
+                                    /* .hasSumGreaterThan(100) */
```

Comment:
can we keep this? (i see the sleep 100 above)

Reply:
Oh, that method is already available in 1.10 sdk-testing - I'll uncomment these. Thanks!